ocrd/ocrd/cli/workspace.py

```diff
@@ -11,6 +13,12 @@
 
 log = getLogger('ocrd.cli.workspace')
 
+def clean_id(dirty):
+    if re.match('^[0-9]', dirty):
+        raise ValueError("Make sure files and directories do not begin with a numeral which will lead to invalid XML identifiers")
+    return re.sub('[^A-Za-z0-9-_]+', '_', dirty)
```

Comment:
I didn't know this restriction of XML ids. (And you sure colons and dots are out?)

How about moving this from ocrd.cli.workspace to ocrd_utils?

Reply:
> I didn't know this restriction of XML ids. (And you sure colons and dots are out?)

METS defines it's `ID` as `xsd:int` which is a restriction on `xsd:NCName` which requires

* first character to be `[A-Za-z]`
* no colons

We ran into this before and it's pretty confounding if you don't know what to look for (in the cases I ran into this, culprit was always purely numeric file names).

Dots would be possible, I tend to avoid them because they're often part of the syntax of query languages like CSS selectors.

Reply:
> How about moving this from ocrd.cli.workspace to ocrd_utils?

This should be checked in regular `add_file()` as well IMO.

Reply:
https://github.com/OCR-D/core/issues/447

Reply:
#500